src/ngraph/op/experimental/dyn_slice.cpp

```diff
@@ -0,0 +1,95 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "ngraph/op/experimental/dyn_slice.hpp"
+#include <memory>
+
+using namespace std;
+using namespace ngraph;
+
+op::DynSlice::DynSlice(const shared_ptr<Node>& arg,
+                       const shared_ptr<Node>& lower_bounds,
+                       const shared_ptr<Node>& upper_bounds,
+                       const shared_ptr<Node>& strides)
+    : Op("DynSlice", check_single_output_args({arg, lower_bounds, upper_bounds, strides}))
+{
+    constructor_validate_and_infer_types();
+}
+
+op::DynSlice::DynSlice(const shared_ptr<Node>& arg,
+                       const shared_ptr<Node>& lower_bounds,
+                       const shared_ptr<Node>& upper_bounds)
+    : Op("DynSlice", check_single_output_args({arg, lower_bounds, upper_bounds}))
+{
+    constructor_validate_and_infer_types();
+}
+
+void op::DynSlice::validate_and_infer_types()
+{
+    // If we only have 3 args, it means strides should have default value 1.
+    if (get_input_size() == 3)
```

Comment:
I'd eliminate the strides-optional constructor and therefore this check.

Reply:
fixed.